src/zsf_config.c

```diff
@@ -41,36 +39,54 @@ int _zsf_ini_handler(char *section, char *key, char *value, void *data_ptr) {
     } else if (!strcmp(key, "sealock_operational_parameters")) {
       config_ptr->locks[lock_index].operational_parameters_file = strdup(value);
     } else if (!strcmp(key, "initial_head_lock")) {
-      config_ptr->locks[lock_index].phase_state.head_lock = strtod(value, &end_ptr);
+      config_ptr->locks[lock_index].phase_state.head_lock = ini_parse_double(value, &status);
     } else if (!strcmp(key, "initial_salinity_lock")) {
-      config_ptr->locks[lock_index].phase_state.salinity_lock = strtod(value, &end_ptr);
+      config_ptr->locks[lock_index].phase_state.salinity_lock = ini_parse_double(value, &status);
     } else if (!strcmp(key, "initial_saltmass_lock")) {
-      config_ptr->locks[lock_index].phase_state.saltmass_lock = strtod(value, &end_ptr);
+      config_ptr->locks[lock_index].phase_state.saltmass_lock = ini_parse_double(value, &status);
     } else if (!strcmp(key, "initial_temperature_lock")) {
       // TODO: Add temperature to lock?
     } else if (!strcmp(key, "initial_volume_ship_in_lock")) {
-      config_ptr->locks[lock_index].phase_state.volume_ship_in_lock = strtod(value, &end_ptr);
+      config_ptr->locks[lock_index].phase_state.volume_ship_in_lock =
+          ini_parse_double(value, &status);
     } else if (!strcmp(key, "lock_length")) {
-      config_ptr->locks[lock_index].parameters.lock_length = strtod(value, &end_ptr);
+      config_ptr->locks[lock_index].parameters.lock_length = ini_parse_double(value, &status);
     } else if (!strcmp(key, "lock_width")) {
-      config_ptr->locks[lock_index].parameters.lock_width = strtod(value, &end_ptr);
+      config_ptr->locks[lock_index].parameters.lock_width = ini_parse_double(value, &status);
     } else if (!strcmp(key, "lock_bottom")) {
-      config_ptr->locks[lock_index].parameters.lock_bottom = strtod(value, &end_ptr);
+      config_ptr->locks[lock_index].parameters.lock_bottom = ini_parse_double(value, &status);
+    } else if (!strcmp(key, "flow_profile")) {
+      int array_length = 0;
+      double *value_array = ini_parse_double_list(value, &array_length, &status);
+      if (array_length && value_array && status == INI_OK) {
```

Comment:
Would it also be possible to check only for != INI_FAIL?

Reply:
Agreed:
First two checks can be removed.
Also: in unit test we need to check handling of empty list.